tools/pkglint/lib/rules.ts

```diff
@@ -173,34 +173,69 @@ export class StabilitySetting extends ValidationRule {
       return;
     }
 
-    let stability = pkg.json.stability;
-    switch (stability) {
-      case 'experimental':
-      case 'stable':
-      case 'deprecated':
-        if (pkg.json.deprecated && stability !== 'deprecated') {
-          pkg.report({
-            ruleName: this.name,
-            message: `Package is deprecated, but is marked with stability "${stability}"`,
-            fix: () => pkg.json.stability = 'deprecated',
-          });
-          stability = 'deprecated';
-        }
-        break;
-      default:
-        const defaultStability = pkg.json.deprecated ? 'deprecated' : 'experimental';
-        pkg.report({
-          ruleName: this.name,
-          message: `Invalid stability configuration in package.json: ${JSON.stringify(stability)}`,
-          fix: () => pkg.json.stability = defaultStability,
-        });
-        stability = defaultStability;
+    // 'stability' contains the default API-level stability, either 'experimental', 'stable',
+    // 'deprecated' or 'external'. This is the value that will be used for every class
+    // unless it is annotated otherwise. It must be one of these values, which will be
+    // enforced by 'jsii'.
+    //
+    // On top of this, we have conceptual library-level "maturity", which is one of
+    // the following stages: 'cfn-only', 'experimental', 'developer-preview', 'stable',
+    // 'deprecated'. These render into stability banners in the README.
+    //
+    // The maturities mostly map onto default stabilities, and so maturity will
+    // be derived from stability if possible. The mapping is:
+    //
+    //     STABILITY      MATURITY
+    //     experimental   cfn-only
+    //                    experimental
+    //                    developer-preview
+    //     stable         stable
+    //     deprecated     deprecated
+    //     external       (not used, we don't allow 'external' as default stability)
+    //
+    // Since 'stability: experimental' maps onto 3 different maturities, you must annotate
+    // this manually. Otherwise, the stability value will be used to determine maturity.
+    //
+    // (Potentially, 'cfn-only' could have been derived automatically, but that would require
+    // making pkglint depend on 'jsii-reflect'... let's not go there).
```

Comment:
Why can't we just check if there are any files under `lib` besides `index.ts`?

Reply:
https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-cloudtrail/lib

![not so quick](https://vignette.wikia.nocookie.net/simpsons/images/e/e9/Nelson_Ha-Ha.jpg/revision/latest/top-crop/width/360/height/360?cb=20121205194057)

Reply:
Because we no consistency :)

And in any case, I think I'd prefer a model in which maturity is leading.